data/blog/boosting-mobile-session-replay-performance-with-view-renderer-v2.md

```diff
@@ -0,0 +1,312 @@
+---
+title: 'Boosting Mobile Session Replay Performance with View Renderer V2'
+date: '2025-07-28'
+tags: ['cocoa', 'session-replay', 'mobile', 'sdk', 'replay']
+draft: false
+summary: 'A detailed analysis of iOS rendering bottlenecks led to a custom solution that achieved 4-5x better performance.'
+images: [/images/boosting-mobile-session-replay-performance-with-view-renderer-v2/hero.jpg]
+layout: PostSimple
+canonicalUrl: boosting-mobile-session-replay-performance-with-view-renderer-v2
+authors: ['philniedertscheider']
+---
+
+After making Session Replay GA for Mobile, the adoption rose quickly and more feedback reached us. In less great news, our Cocoa SDK users reported that the performance overhead of Session Replay on older iOS devices **made their apps unusable**.
+
+So we went on the journey to find the culprit and found a solution that yielded **4-5x better performance** in our benchmarks 🔎📈
+
+![Session Replay on Mobile](/images/boosting-mobile-session-replay-performance-with-view-renderer-v2/hero.jpg)
+
+To understand what's happening under the hood of Session Replay on Mobile, let's first explore how mobile screen recording works before diving into the technical details.
+
+# Frame Rates In A Nutshell
+
+A screen recording is a video made up of many fast-displayed images called `frames`. Human eyes can process about 60 frames per second (`frame rate`), measured in hertz (`1 Hz` = `1 unit per second`), creating the illusion of a moving image. The frame rates vary by use case, from `24 Hz` for cinema to `144 Hz` for gaming computer displays.
+
+Higher frame rates make videos smoother, but have downsides:
+
+- More storage and network bandwidth required
+- Less time to capture per frame available, requiring better hardware
+
+When reducing the frame rate to a minimum, the recordings look like **stop motion videos**, with a great example for this style visible in this [YouTube video](https://www.youtube.com/watch?v=MEglOulvgSY). The frames are just sequential photos, yet they still feel like a moving photo — essentially a video.
+
+**This is essentially what we are doing with Session Replay on Mobile!**
+
+Instead of having a full screen recording, which is resource-heavy, we capture a screenshot every second instead. Every 5 seconds, these frames are combined into a video segment, creating a stop-motion-like recording. These video segments are then uploaded to Sentry and stitched into a full session replay.
```

Comment:
"Instead of having a full screen recording, which is resource-heavy, we capture a screenshot every second instead."

Now I want to know the `hertz` equivalent measurement for Session Replay :D